R/osf_mv.R

```diff
@@ -59,9 +59,10 @@ osf_mv.osf_tbl_file <- function(x, to, overwrite = FALSE, verbose = FALSE) {
 #' @noRd
 #' @references
 #' https://waterbutler.readthedocs.io/en/latest/api.html#actions
-.wb_file_move <- function(x, to, action, overwrite, verbose) {
+.wb_file_action <- function(x, to, action, overwrite, verbose) {
   action <- match.arg(action, c("move", "copy"))
-  conflict <- ifelse(overwrite, "replace", "warn")
+  if (action == "move") conflict <- ifelse(overwrite, "replace", "warn")   #move
+  if (action == "copy") conflict <- "keep"                                 #copy
```

Comment:
This would mean the `overwrite` argument is ignored.

To be consistent with `osf_mv()`, I think `overwrite = FALSE` should be the default to let users know a conflicting file with the same name already exists at the specified destination.

Also not sure about the usefulness of `"keep"`... Do you think people ever purposely intend to duplicate files in the same location with numbered suffixes? My sense is that's usually done accidentally but I could be overlooking something...

Thoughts?

Reply:
You make a good point.  I had figured to keep the same functionality as the API which sets conflict="keep".  I agree it's not common to make a copy in the same location but OSx has the same behavior of adding a suffix on a file copied to the same location.

We could set the default value for overwrite to throw a warning.  But allow the creation of a copied file (with suffix) as the other option.  The weirdness would be that neither is an 'overwrite'.  The other option is to drop this argument entirely for `osf_cp` and use the 'keep' value for conflict.

Reply:
Good call re OS X. That got me thinking, the Finder also offers an option to *Duplicate* a file if you right click on it. I think that's a more descriptive/explicit name for what the "keep" option does. 

My vote is for `osf_cp`'s default to be `overwrite = FALSE`, which should set the `conflict` variable to `"warn"`.  If the user sets `overwrite = TRUE` then `conflict = "replace"` and the will is overwritten. 

If they *really* want to create a duplicate version of a file or directory we can add something like `osf_duplicate()` for that express purpose.

Reply:
You propose a clean solution.  I'll modify `osf_cp`.  Possibly put together `osf_duplicate` in a separate PR.

Reply:
A PR for `osf_duplicate()` would be great.